packages/astro/src/server/build-clerk-hotload-script.ts

```diff
@@ -0,0 +1,64 @@
+import { createDevOrStagingUrlCache, parsePublishableKey } from '@clerk/shared/keys';
+import { isValidProxyUrl, proxyUrlToAbsoluteURL } from '@clerk/shared/proxy';
+import { addClerkPrefix } from '@clerk/shared/url';
+import type { APIContext } from 'astro';
+
+import { versionSelector } from '../internal/utils/versionSelector';
+import { getSafeEnv } from './get-safe-env';
+
+const { isDevOrStagingUrl } = createDevOrStagingUrlCache();
+
+type BuildClerkJsScriptOptions = {
+  proxyUrl: string;
+  domain: string;
+  clerkJSUrl?: string;
+  clerkJSVariant?: 'headless' | '';
+  clerkJSVersion?: string;
+  publishableKey: string;
+};
+
+const clerkJsScriptUrl = (opts: BuildClerkJsScriptOptions) => {
```

Comment:
This file is mostly duplicative of the logic in clerk-react right? What would it take to re-use the logic from there?

Reply:
This will be address with ECO-4, but thanks for mentioning it